### DISCUSSION TITLE
WP graphql, make fields non nullable

### DISCUSSION THREAD

**Discussion by akzhy** [gatsbyjs/gatsby#28310] [Help] [answered]
Hi,
I am working on a gatsby project that uses typescript and codegen for creating graphql types.
Currently most of the fields from the WordPress side are marked as nullable and because of that when it's type is created these fields are marked as follows

```
title: string | undefined
```
(and so on)

And now I explicitly have to write condition for every field. I know that it's a good thing to ensure all the data is present, but I've already made the required conditions in the WordPress side. 

I've tried to modify the schema as follows (I don't know if it's the right way to do it)

```
exports.createResolvers = ({ createResolvers }) => {
    const resolvers = {
        WpPost: {
            title: {
                type: 'String!'
            }
        },
    }
    createResolvers(resolvers)
}
```

But that didn't do anything. 

So how can I customize the schema so that all or some fields are made non nullable. Or is it something that I need to do from WordPress side? I am using `gatsby-source-wordpress-experimental`

**Comment by LekoArts** [comment, marked as answer]
Hi!

You'll need to use the `createSchemaCustomization` API, not the `createResolvers` one: https://www.gatsbyjs.com/docs/schema-customization/#opting-out-of-type-inference

**Reply by akzhy** [reply]
Thanks, been working on this for hours. 

```
exports.createSchemaCustomization = ({ actions }) => {   
    const { createTypes } = actions
    const typeDefs = `#graphql
        type WpPost implements Node {
            title: String!
        }
    `
    createTypes(typeDefs)
}
```
